gossip/src/crds_gossip_pull.rs

```diff
@@ -53,8 +53,6 @@ use {
 pub const CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS: u64 = 15000;
 // Retention period of hashes of received outdated values.
 const FAILED_INSERTS_RETENTION_MS: u64 = 20_000;
-// Maximum number of pull requests to send out each time around.
-const MAX_NUM_PULL_REQUESTS: usize = 1024;
```

Comment:
why do this refactor and filter sampling instead of reducing number of `MAX_NUM_PULL_REQUESTS`

Reply:
The `MAX_NUM_PULL_REQUESTS` is just an upper bound on the number of outgoing requests regardless of crds table size and it is not proportional. The sampling is proportional to the number of requests going out originally.